### ISSUE TITLE
New v3 path error message should distinguish between "lack of key" and "key equals null"

### ISSUE THREAD

**Issue by zerkms** [azjezz/psl#494]
**Is your feature request related to a problem? Please describe.**

```php
Type\shape(['foo' => Type\string()], true)->assert([]);
Type\shape(['foo' => Type\string()], true)->assert(['foo' => null]);
```

Both these two evaluations result in a `AssertException` with a `Expected "array{'foo': string}", got "null" at path "foo".` message.

It's misleading as the lack of the key is definitely different beast to the key that is `= null`.

**Describe the solution you'd like**
I would prefer two distinct messages (non-native speaker here, so needs polishing):

```
Expected "array{'foo': string}", path "foo" does not exist.
Expected "array{'foo': string}", got "null" at path "foo".
```

**Describe alternatives you've considered**

**Additional context**

**Comment by veewee** [comment]
Hi @zerkms,

That sure is possible. Care to give it a PR?
Related exception lines in code:

* https://github.com/azjezz/psl/blob/6683d60ba4d3abf8b4a4a9d94c39d0d08caa0214/src/Psl/Type/Internal/ShapeType.php#L147
* https://github.com/azjezz/psl/blob/6683d60ba4d3abf8b4a4a9d94c39d0d08caa0214/src/Psl/Type/Internal/ShapeType.php#L199

**Comment by zerkms** [comment]
> Related exception lines in code:

Yep, I already debugged through it :-) It was also to confirm whether the corresponding change would be accepted/welcomed overall.
